packages/superset-ui-legacy-plugin-chart-table/src/ReactDataTable.tsx

```diff
@@ -116,13 +114,13 @@ export default function ReactDataTable(props: DataTableProps) {
     if (typeof val === 'string') {
       return filterXSS(val, { stripIgnoreTag: true });
     }
-    if (isPercentMetric(key)) {
+    if (percentMetricsSet.has(key)) {
       // in case percent metric can specify percent format in the future
       return formatNumber(format || PERCENT_3_POINT, val as number);
     }
-    if (isMetric(key)) {
+    if (metricsSet.has(key)) {
       // default format '' will return human readable numbers (e.g. 50M, 33k)
-      return formatNumber(format || '', val as number);
+      return formatNumber(format || SMART_NUMBER, val as number);
```

Comment:
no need to explicitly set `SMART_NUMBER` here. the `NumberFormatterRegistry` has ability to set a fallback if the format string is not known by the registry. The fallback is `SMART_NUMBER` by default. 

Not hardcoding it here will make it possible to override fallback at the registry.

Reply:
After installing all node_modules fresh, if I pass `''`, I will get a `this.get(...) is not a function` error from https://github.com/apache-superset/superset-ui/blob/9bb5c471e0e3d05da38929feb00ade7d0ecdc474/packages/superset-ui-number-format/src/NumberFormatterRegistry.ts#L42 . But if I use `formatNumber(format, val as number)`, I will get "type not supported" error, because `formatNumber` does not accept `undefined`: https://github.com/apache-superset/superset-ui/blob/9bb5c471e0e3d05da38929feb00ade7d0ecdc474/packages/superset-ui-number-format/src/NumberFormatterRegistrySingleton.ts#L12

Reply:
My bad. Let me fix the bug in `@superset-ui/number-format` real quick. Give 15mins.